### ISSUE TITLE
Wording for "license" section?

### ISSUE THREAD

**Issue by saibotsivad** [zcei/standard-readme#4]
I'm wondering if there's any standardization on the wording for the "License" section. I want it to be legally clear, so the lawyer folk will read it and be comfortable. A few I've tried:

```
# License
[MIT](/LICENSE)

# License
All contents of this repo release under the [MIT](/LICENSE).

# License
All code and documentation release under the [MIT](/LICENSE).
```

Are there any big projects to reference that are being used by big beauracratic organizations? They probably have lawyer-checked their licensing.

**Comment by saibotsivad** [comment]
Looks like a big project like [AngularJS](https://github.com/angular/angular.js) doesn't even mention a license in their readme.

**Comment by wooorm** [comment]
**I’M  NOT A LAWYER**

From what I remember, the LICENSE file in your project root is much more important, and a LICENSE section in the README has no legal impact. however, IMHO adding short, detailed info in the README is really useful.
- [react](https://github.com/facebook/react#license) has a section;
- [bootstrap](https://github.com/twbs/bootstrap#copyright-and-license) too;
- Node has no license in README;
- jQuery has no license in README;
- airbnb/javascript includes a whole `LICENSE` file in the README (they do not have the file)
- ...

So I guess you can say there’s no real standard; most use a different format.

**Comment by wooorm** [comment]
Maybe @kemitchell might be of help here?

**Comment by kemitchell** [comment]
@wooorm, pleased to make your acquaintance!

Unfortunately, I don't think there is a single, "right" legal answer here, even just considering US law. (And that's hardly fair to open-source friends abroad!) You seem to have found that out yourself, and I was very interested and grateful to read your small "survey" of what some very important projects are doing.

More practically, I really can't be on the hook for legal advice to users of `standard-readme`. Folks who need legal advice should engage an attorney who's responsible to them.

In case you or others want to do some research for yourselves, under US the law, the question of how to make software license terms stick is related to the question of when other take-it-or-leave-it terms, like website terms of use, actually create contracts. Search terms you might try include "affirmative assent", "contract consideration", "meeting of the minds", and "contract of adhesion". For a different take on the problem, you might read some of what Eben Moglen of the Software Freedom Law Center has written on the GPL as a license as opposed to a contract.

For reading on software licenses more generally, I highly recommend Larry Rosen's introductory _Open Source Licensing_, which is available online.

**Comment by wooorm** [comment]
@kemitchell, likewise :)

I was hoping that you could reassure me/us that there might be a simple answer here, but as I already thought, and as you affirmed, there is none.

Thanks for the suggestions, I'll read the book you recommended, and hope to find some useful practical info there! I'll also check out your search tips.

---

Back to regularly READMEs, I personally like to have a license in two places, README and LICENSE, because they might be extracted by tooling and shown without context. Also, I link from the license section in my READMEs to the LICENSE file for the full document.

And, in addition, I always add a copyright notice next to the license identifier in the README, because IMHO copyright I what denies others to use and copy a project, and the license lifts some limitations.

To summarise what I do in my README, I use `MIT © Titus Wormer`, where MIT links to the LICENSE file, and my name to my website.

**Comment by zcei** [comment]
As the default copyright laws apply even without stating so explicitly by e.g. `MIT © Titus Wormer`, I would like to introduce a block, that - if left out - gives a warning, where you use the "human readable" version of how you can use the package and link to the "lawyer readable" version.

GitHub also [states here](https://help.github.com/articles/open-source-licensing/) that a`LICENSE` file is all you need - and IMHO they wouldn't do so, if it's really such an ambiguous topic.

In the end the law perspective for me comes down to whether the license is obviously visible to a visitor.
For me it is, as the license file on root level is showed while scrolling, even before hiting the readme section describing the license.

**Why a block in readme at all then?** Be nice to people, for me a sentence about the license is good-practice, because no one wants to read through a `LICENSE` file
